lib/web_ui/lib/src/engine/renderer.dart

```diff
@@ -38,6 +38,19 @@ abstract class Renderer {
         useCanvasKit = FlutterConfiguration.useSkia;
       }
 
+      // Warn users in development that anything other than canvaskit is deprecated.
+      assert(() {
+        if (!useCanvasKit) {
```

Comment:
Will this trigger if you pick `auto` but are debugging in desktop Chrome (I imagine it's the most common case)? I think we want `if (['html', 'auto'].contains(configuration.requestedRendererType))`.

Reply:
In the current branch of the if, skwasm has already been chosen, and when we hit this we know that the HTML renderer is being selected, we're trying to reconstruct why.

I think the only corner case that this won't pick up at runtime is if the user has set `--web-renderer=auto` but then used `renderer: canvaskit` in their [load() call](https://docs.flutter.dev/platform-integration/web/initialization#the-_flutter-loader-load-api).

Reply:
What is the value of `useCanvasKit` when you do `--web-renderer=auto` and test in Chrome?

Reply:
> What is the value of useCanvasKit when you do --web-renderer=auto and test in desktop Chrome?

`true` (as expected, no?)

Reply:
I think we could've done:
```dart
if (FlutterConfiguration.flutterWebAutoDetect || !useCanvasKit) {
```
to capture all cases we want to warn against.